compiler/passes/docs.cpp

```diff
@@ -500,8 +519,23 @@ void printFunction(std::ofstream *file, FnSymbol *fn, bool method) {
     if (fn->isSecondaryMethod()) {
       if (fn->numFormals() > 1) {
         ArgSymbol *myTypeHolder = fn->getFormal(2);
-        if (myTypeHolder->hasFlag(FLAG_ARG_THIS))
-          *file << myTypeHolder->type->symbol->name << ".";
+        if (myTypeHolder->hasFlag(FLAG_ARG_THIS)) {
+          Expr *typeExpr = myTypeHolder->typeExpr;
+          if (BlockStmt *body = toBlockStmt(typeExpr)) {
+            if (UnresolvedSymExpr *typeName = toUnresolvedSymExpr(body->body.tail)) {
+              *file << typeName->unresolved << ".";
+            } else {
+              INT_ASSERT(false);
```

Comment:
The outer one should not be an assert (as the function is used for both secondary and primary methods, as well as functions that aren't methods at all), but it seems reasonable to me to convert the other ones.